pixel-saver@deadalnix.me/extension.js

```diff
@@ -44,20 +44,36 @@
  * [5]:http://www.webupd8.org/2011/05/how-to-remove-maximized-windows.html
  *
  */
+const Lang = imports.lang;
 const ExtensionUtils = imports.misc.extensionUtils;
 const Me = ExtensionUtils.getCurrentExtension();
 const Decoration = Me.imports.decoration;
 const Buttons = Me.imports.buttons;
 const AppMenu = Me.imports.app_menu;
+const Settings = Me.imports.settings;
+
+let options;
 
 function init(extensionMeta) {
 	Buttons.init(extensionMeta);
 	Decoration.init(extensionMeta);
 	AppMenu.init(extensionMeta);
 }
 
+function loadOptions() {
+	options = new Settings.Options();
+	options.BUTTON_POSITION.changed(Lang.bind(this, function() {
+		if (options.BUTTON_POSITION.get() == Buttons.Position.HIDDEN) {
+			Buttons.hideButtons();
+		} else {
+			Buttons.showButtons();
+		}
+	}));
+}
+
 function enable() {
-	Buttons.enable();
+	loadOptions();
+	Buttons.enable(options);
```

Comment:
See https://github.com/deadalnix/pixel-saver/pull/56#discussion_r61685799

I leave it up to you decide. Both approaches certainly work.